pkg/kubectl/cmd/expose.go

```diff
@@ -182,6 +182,17 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 			params["ports"] = strings.Join(ports, ",")
 		}
 	}
+
+	// Always try to derive protocols from the exposed object, as
+	// user may specify UDP protocol
+	protocolsMap, err := f.ProtocolsForObject(inputObject)
```

Comment:
protocol in `params` will be always found for generators that require a protocol such as the service generators. Doesn't have to do anything with the default value in the flag.